Improve attr_implements_fp.swift test.

rudkx opened the pull request:
Several small changes:

- Ensure that it tests calling both variants from generic contexts.
- Use a new protocol rather than Comparable so that we can ensure this works properly with designated types
- Restate the requirement for comparison as a @_nonoverride.
- Actually make use of designated types, ensuring that we produce the expected result.

Commit:
Add another test for @_implements when calling from a generic context.

Commit:
Introduce Cauxmparable in place of Comparable in @_implements test.

Commit:
Rename Cauxmparables requirement from '<' to '.<' in an @_implements test.

Commit:
Restate the .< requirement as a @_nonoverride in @_implements test.

Commit:
Add new @_implements tests for calling through another generic context.

Commit:
Use designated types in test of @_implements, to ensure we maintain the expected behavior.

rudkx commented:
@swift-ci Please smoke test

Commit:
Move frontend argument overrides earlier.

rudkx commented:
@swift-ci Please smoke test

jrose-apple commented on test/attr/attr_implements_fp.swift:
```diff
@@ -1,6 +1,6 @@
 // RUN: %empty-directory(%t)
 // RUN: echo 'main()' >%t/main.swift
-// RUN: %target-swiftc_driver -o %t/a.out %s %t/main.swift -Xfrontend -enable-operator-designated-types -Xfrontend -solver-enable-operator-designated-types
+// RUN: %target-swiftc_driver -Xfrontend -enable-operator-designated-types -Xfrontend -solver-enable-operator-designated-types -o %t/a.out %s %t/main.swift
```
If you're improving this test anyway, please use `%target-build-swift` instead of `%target-swiftc_driver`, which includes things like `-module-cache-path`.

jrose-apple replied:
(Okay, that one specifically is in both, but there are some other options that are only in `target-build-swift`. And I should probably look into whether we still need both.)

rudkx commented:
Interesting. Swapping the position of the frontend arguments doesn't make any difference, but running tests with `build-script` rather than `lit` does (the latter passing, which is how I've been testing it).

Commit:
Revert "Move frontend argument overrides earlier."

This reverts commit 0346fedf6547d58ffb181b755884ee3a881299c9.

It turns out moving these arguments didn't result in this test
passing. For some reason it's failing with `build-script -t`, but
passing with `lit`.

Commit:
Switch to `%target-build-swift`.

Commit:
Also update `attr_implements_serial.swift`.

This also builds `attr_implements_fp.swift`, and then uses it, so it
also needed updating.

rudkx commented:
Oh actually it's `attr_implements_serial.swift` that is failing.

rudkx commented:
@swift-ci Please smoke test
